### DISCUSSION TITLE
linode blocks smtp, so pocketbase unusable

### DISCUSSION THREAD

**Discussion by garlandcrow** [pocketbase/pocketbase#1064] [Q&A] [answered]
Forgot about that and now burned after porting over a site I was working on to pocketbase. I always used APIs for sendgrid/mailgun in the past and now I remember why....

**Comment by benallfree** [comment]
From their faq https://www.linode.com/community/questions/19082/i-just-created-my-first-linode-and-i-cant-send-emails-why

> If you've recently signed up and need to send email, [please open a ticket from the Cloud Manager](https://cloud.linode.com/support/tickets) and provide us with the following information…

So it seems that they just block by default.

**Comment by pchasco** [comment, marked as answer]
This is a common and reasonable restriction. They don't want their IP's to end up on mail blacklists.

**Reply by garlandcrow** [reply]
Yea, it's not an unreasonable restriction for Linode, its a poor choice IMO for Pocketbase to only use SMTP for emails when the majority of people that are going to use it are new/small projects, and most of those are going to run on some shared instance on DO/Linode just to have their SMTP requests blocked. Just more friction to get something up and running vs something like setting API keys to mailgun/sendgrid in some settings menu and not worrying about it.

So likely I will just have to write my own mail handling and not use the Pocketbase integrated one so I can continue to evaluate it. Waiting and hoping I get approved to send email verifications to new signups for every project I make is just not something I will do. So now a 2nd disappointing thing for something I was hoping was a nice fit for new/small projects.

**Reply by ganigeorgiev** [reply]
@garlandcrow There are too many different mail providers and it will be too hard to maintain if we have to integrate directly with each unique API. Where SMTP on the other hand is probably already supported by every public email service.

As the other users mentioned, restricting outbound 25/465/587/etc. ports is a common practice by vps providers. Hetzner for example I think blocks 25 and 465 for 1 month for new accounts and has similar requirements as Linode. For DigitalOcean and Vultr I don't think they have such limitation but they are also commonly used by spam and malicious bots.

_Update: Vultr and DO also blocks 25 port based on https://docs.digitalocean.com/support/why-is-smtp-blocked/ and https://www.vultr.com/docs/what-ports-are-blocked/_

**Reply by pchasco** [reply]
I couldn’t tell whether it was possible in the current version of PocketBase, but it might be useful if there were an event hook that is called before any email is to be sent. The hook will have all required parameters to send the email with an alternative service. If the hook decides, it can communicate back to PocketBase that the email has been handled and PocketBase would not attempt to send it via smtp.

**Reply by ganigeorgiev** [reply]
@pchasco You can use the `OnMailerBefore*` hooks and intercept the default email send functionality by returning `hook.StopPropagation` - https://pocketbase.io/docs/event-hooks/#mailer-hooks (there are examples if you expand the corresponding event hook)

**Reply by pchasco** [reply]
Thanks! I was thinking more along the lines of a single hook that is used for all emails instead of having to implement several hooks. However, no problem if the current state is how it has to be!
